src/realm/sync/instruction_replication.cpp

```diff
@@ -750,7 +761,22 @@ void SyncReplication::populate_path_instr(Instruction::PathInstruction& instr, c
 {
     REALM_ASSERT(key);
     // The first path entry will be the column key
-    REALM_ASSERT(path[0].is_col_key());
+    std::string field_name;
+    if (path[0].is_col_key()) {
+        auto ck = path[0].get_col_key();
+        if (table.is_additional_props_col(ck)) {
+            REALM_ASSERT(path.size() > 1);
+            field_name = path[1].get_key();
+            // Erase the "__additional__" part of the path
```

Comment:
should we assert that that's the case?

Reply:
No need. The condition in line 767 ensures that.